### ISSUE TITLE
Ability to clear cache of memoized functions

### ISSUE THREAD

**Issue by jiffyclub** [pytoolz/toolz#228]
Function's decorated with functools' [lru_cache](https://docs.python.org/3.4/library/functools.html#functools.lru_cache) have a `cache_clear()` function attached to them so that the cache can be manually invalidated. It'd be useful to me if toolz's `memoize` function did the same thing.

**Comment by eriknw** [comment]
Yeah, it should definitely be possible to clear a cache.  I think it is sufficient to expose `func`, `cache`, and `key` as attributes of the memoized function.

Unlike `lru_cache`, `memoize` can accept a cache object from the user.  Giving the user direct access to that is more flexible.

**Comment by xrr** [comment]
It would also be useful to do a "cache refresh", on a given call, ie.:
1. clear from cache specifically the key associated to the given call
2. do a "regular call" (obtain result from decorated function, save it into the cache & return it)
